README.md

```diff
@@ -1,2 +1,79 @@
-# beacon-starter
+# Beacon Reader Example
 
+> A starter project for reading beacon values from a smart contract
+
+This project is composed of the following:
+
+1. A test that uses a mocked RrpBeaconServer to simulate reading a beacon value from a smart contract.
+1. A script to deploy the BeaconReaderExample contract to any network. This script will set the address of the RrpBeaconServer contract deployed on the selected network and after the script finishes it will display the address of the contract deployed but also save this address to `deployments` directory on the root of this repo. If selected network is not `hardhat` or `localhost` then this address must be then whitelisted by the RrpBeaconServer manager account prior to reading the beacon value.
+
+## Instructions
+
+1. Clone this repo
+1. Run the following to install the dependencies
+
+```sh
+npm install
+```
+
+Now depending on the network you are using you will need to do the following:
+
+### Default network
+
+Simplest way to test this project is by using the integrated [Hardhat network](https://hardhat.org/hardhat-network/) and not extra steps are needed in this case to run both the test and the deployment script.
+
+#### Test
+
+```sh
+hardhat test test/BeaconReaderExample.test.js
+```
```

Comment:
After running `npm install` then this command I got an error:
```sh
$ hardhat test test/BeaconReaderExample.test.js
hardhat: command not found
```
The same command prefixed with `npx` did run (I had to prefix all commands with `npx` from here forward), but errored:
```sh
$ npx hardhat test test/BeaconReaderExample.test.js
Error HH8: There's one or more errors in your config file:

  * Invalid value undefined for HardhatConfig.networks.undefined.url - Expected a value of type string.
  * Invalid value {} for HardhatConfig.networks.undefined - Expected a value of type HttpNetworkHDAccountsConfig.
```
which is because the `hardhat.config.js` has interpolated values from `.env`, which isn't present, just `.env.example`. Consequently I feel there should be instructions regarding how to prepare the `.env` file for each network.

This issue actually applies to all `hardhat` commands at the moment because the config file appears to always get read.

I created a `.env` file then this worked.

Reply:
I'm going to make interpolation more resilient in order to only require creating a .env when user wants to use a remote network (other than hardhat or localhost)